frontend/apps/crates/entry/jig/edit/src/edit/sidebar/module/actions.rs

```diff
@@ -157,6 +153,23 @@ pub fn assign_kind(state: Rc<State>, kind: ModuleKind) {
                     let mut modules = state.sidebar.modules.lock_mut();
                     modules.remove(index);
                     modules.insert_cloned(index, module);
+
+                    // Only add a new placeholder module once the above request has completed and
+                    // the new module has been added to the list of modules.
+                    let placeholder_exists = {
+                        match modules.last() {
+                            // If the list of modules is not empty and the last module is None, then it is
+                            // a placeholder module.
+                            Some(module) => module.is_none(),
```

Comment:
`module` in this case is `Rc<Option<_>>` so we can't use a pattern like `Some(None)`